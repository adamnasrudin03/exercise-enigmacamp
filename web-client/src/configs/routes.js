
import { AboutPage } from '../pages/about';
import { ItemsPage, ItemPage } from '../pages/item';
import { StocksPage, StockPage, StockSummaryPage } from '../pages/stock';
import { TransactionsPage, TransactionPage, TransactionSummaryPage } from '../pages/transaction';
import { UnitsPage, UnitPage } from '../pages/unit';

import { ErrorsPage } from "../pages/error";


const routes = [
    {
        path: '/',
        component: TransactionsPage,
        exact: true
    },
    {
        path: '/items',
        component: ItemsPage
    },
    {
        path: '/items/add',
        component: ItemPage
    },
    {
        path: '/items/:id',
        component: ItemPage
    },
    {
        path: '/units',
        component: UnitsPage
    },
    {
        path: '/units/add',
        component: UnitPage
    },
    {
        path: '/units/:id',
        component: UnitPage
    },
    {
        path: '/stocks',
        component: StocksPage
    },
    {
        path: '/stocks/summary',
        component: StockSummaryPage
    },
    {
        path: '/stocks/add',
        component: StockPage
    },
    {
        path: '/stocks/:id',
        component: StockPage
    },
    {
        path: '/transactions',
        component: TransactionsPage
    },
    {
        path: '/transactions/summary',
        component: TransactionSummaryPage
    },
    {
        path: '/transactions/add',
        component: TransactionPage
    },
    {
        path: '/transactions/:id',
        component: TransactionPage
    },

    {
        path: '/about',
        component: AboutPage,

    },
    {
        path: '*',
        component: ErrorsPage,
        props: {
            code: 404
        }
    }
];

export default routes;