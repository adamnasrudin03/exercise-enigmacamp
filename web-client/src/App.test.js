import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

import routes from './configs/routes';


describe('App', () => {

  const wrapper = shallow(<App />);

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();

  })

  it('should have N route(s)', () => {
    expect(wrapper.find("Route")).toHaveLength(routes.length);

  });

});