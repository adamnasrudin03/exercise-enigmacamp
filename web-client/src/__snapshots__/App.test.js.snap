// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      key="0"
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      key="1"
      path="/items"
      render={[Function]}
    />
    <Route
      exact={true}
      key="2"
      path="/items/add"
      render={[Function]}
    />
    <Route
      exact={true}
      key="3"
      path="/items/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      key="4"
      path="/units"
      render={[Function]}
    />
    <Route
      exact={true}
      key="5"
      path="/stocks"
      render={[Function]}
    />
    <Route
      exact={true}
      key="6"
      path="/transactions"
      render={[Function]}
    />
    <Route
      exact={true}
      key="7"
      path="*"
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
